CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+$ composer cs-fix
+```
+
+If the above fixes any CS issues, please re-run the tests to ensure they pass, and make sure you add and commit the changes after verification.
+
+## Recommended Workflow for Contributions
+
+Your first step is to establish a public repository from which we can pull your work into the master repository.
+We recommend using [GitHub](https://github.com), as that is where the component is already hosted.
+
+1. Setup a [GitHub account](http://github.com/), if you haven't yet
+2. Fork the repository (http://github.com/bitExpert/disco)
+3. Clone the canonical repository locally and enter it.
+
+```console
+$ git clone git://github.com:bitExpert/disco.git
+$ cd disco
+```
+
+4. Add a remote to your fork; substitute your GitHub username in the command
+   below.
+
+```console
+$ git remote add {username} git@github.com:{username}/disco.git
+$ git fetch {username}
+```
+
+### Keeping Up-to-Date
+
+Periodically, you should update your fork or personal repository to match the canonical Disco repository.
+Assuming you have setup your local repository per the instructions above, you can do the following:
+
+```console
+$ git checkout master
+$ git fetch origin
+$ git rebase origin/master
+# OPTIONALLY, to keep your remote up-to-date -
+$ git push {username} master:master
+```
+
+If you're tracking other branches — for example, the "develop" branch, where new feature development occurs — you'll want to do the same operations for that branch; simply substitute  "develop" for "master".
+
+### Working on a patch
+
+We recommend you do each new feature or bugfix in a new branch.
+This simplifies the task of code review as well as the task of merging your changes into the canonical repository.
+
+A typical workflow will then consist of the following:
+
+1. Create a new local branch based off either your master or develop branch.
+2. Switch to your new local branch. (This step can be combined with the
+   previous step with the use of `git checkout -b`.)
+3. Do some work, commit, repeat as necessary.
+4. Push the local branch to your remote repository.
+5. Send a pull request.
+
+The mechanics of this process are actually quite trivial.
+Below, we will create a branch for fixing an issue in the tracker.
+
+```console
+$ git checkout -b hotfix/9295
+Switched to a new branch 'hotfix/9295'
+```
+
+...do some work...
+
+```console
+$ git commit
+```
+
+...write your log message...
+
+```console
+$ git push {username} hotfix/9295:hotfix/9295
+Counting objects: 38, done.
+Delta compression using up to 2 threads.
+Compression objects: 100% (18/18), done.
+Writing objects: 100% (20/20), 8.19KiB, done.
+Total 20 (delta 12), reused 0 (delta 0)
+To ssh://git@github.com/{username}/disco.git
+   b5583aa..4f51698  HEAD -> master
+```
+### Sending a PR
+
+Before sending a PR, run `composer.phar validate`, as the Travis build will also run the command and will fail if the composer.lock file is not up-to-date with the changes in the composer.json file.
```

Comment:
It should be mentioned that the CodeSniffer tests as well the PHPUnit tests need also to run.